cloud/aws/services/ec2/vpc.go

```diff
@@ -14,44 +14,85 @@
 package ec2
 
 import (
-	"errors"
 	"fmt"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"sigs.k8s.io/cluster-api-provider-aws/cloud/aws/providerconfig/v1alpha1"
 )
 
 type vpcs interface {
-	DescribeVpcs(*ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error)
+	DescribeVpcs(input *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error)
+	CreateVpc(input *ec2.CreateVpcInput) (*ec2.CreateVpcOutput, error)
+	DeleteVpc(input *ec2.DeleteVpcInput) (*ec2.DeleteVpcOutput, error)
 }
 
 type VPC struct {
-	ID string
+	ID        string
+	CidrBlock string
 }
 
-func (s *Service) ReconcileVPC(id string) (*VPC, error) {
+func (s *Service) ReconcileVPC(v *v1alpha1.VPC) (*VPC, error) {
```

Comment:
This is a pointer and the interface wants a value, i'm more inclined to pass the pointer around unless there is good reason to pass-by-value, but i don't see it here

Reply:
I left it as a value anywhere, I'm ok passing it as a reference though.